lightning/src/offers/offer.rs

```diff
@@ -204,13 +238,36 @@ impl OfferBuilder {
 			}
 		}
 
+		Ok(self.build_without_checks())
+	}
+
+	fn build_without_checks(mut self) -> Offer {
+		// Create the metadata for stateless verification of an InvoiceRequest.
+		if let Some(metadata_material) = self.metadata.material_mut() {
+			let mut tlv_stream = self.offer.as_tlv_stream();
+			tlv_stream.metadata = None;
+			tlv_stream.node_id = None;
+			tlv_stream.write(metadata_material).unwrap();
+
+			if self.offer.paths.is_none() {
+				self.metadata = Metadata::Derived(metadata_material.clone());
+			}
+		}
+
+		let (metadata, signing_pubkey) = self.metadata.into_parts();
+
+		if let Some(metadata) = metadata {
+			self.offer.metadata = Some(metadata);
```

Comment:
I can't find the resolution to [this discussion](https://github.com/lightningdevkit/rust-lightning/pull/1989#issuecomment-1457185176). Why do we need to also set the metadata if we're deriving the signing pubkey?